src/templates/common/medallia/index.test.tsx

```diff
@@ -20,8 +15,78 @@ jest.mock('@/lib/utils/medallia', () => ({
   setWindowVaSurvey: jest.fn(),
 }))
 
+// Mock recordEvent
+jest.mock('@/lib/analytics/recordEvent', () => ({
+  recordEvent: jest.fn(),
+}))
+
+interface MedalliaEvent extends CustomEvent {
+  detail: {
+    Form_Type?: string
+    Form_ID?: string
+    Feedback_UUID?: string
+    Content?: string
+  }
+}
+
+const mockEvent: MedalliaEvent = {
+  detail: {},
+  initCustomEvent: function (
+    type: string,
+    bubbles?: boolean,
+    cancelable?: boolean,
+    detail?: unknown
+  ): void {
+    throw new Error('Function not implemented.')
+  },
+  bubbles: false,
+  cancelBubble: false,
+  cancelable: false,
+  composed: false,
+  currentTarget: undefined,
+  defaultPrevented: false,
+  eventPhase: 0,
+  isTrusted: false,
+  returnValue: false,
+  srcElement: undefined,
+  target: undefined,
+  timeStamp: 0,
+  type: '',
+  composedPath: function (): EventTarget[] {
+    throw new Error('Function not implemented.')
+  },
+  initEvent: function (
+    type: string,
+    bubbles?: boolean,
+    cancelable?: boolean
+  ): void {
+    throw new Error('Function not implemented.')
+  },
+  preventDefault: function (): void {
+    throw new Error('Function not implemented.')
+  },
+  stopImmediatePropagation: function (): void {
+    throw new Error('Function not implemented.')
+  },
+  stopPropagation: function (): void {
+    throw new Error('Function not implemented.')
+  },
+  NONE: 0,
+  CAPTURING_PHASE: 1,
+  AT_TARGET: 2,
+  BUBBLING_PHASE: 3,
+}
+
 describe('MedalliaAssets Component', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
   test('renders Script component with correct props', () => {
+    // Mocking useEffect
+    jest.mock('react', () => ({
+      ...jest.requireActual('react'),
+      useEffect: jest.fn(),
+    }))
```

Comment:
😕 Can we just remove this bit if we're redefining `process.env.NEXT_PUBLIC_BUILD_TYPE`? Or does redefining that environment variable not propagate through to the rendered component?

...and...why do you not have to mock it for your newly-added test below? Seems like we don't actually need this bit...or am I missing something? 🤔

Reply:
That is a good point, I was trying to leave un changed things intact. I'll test without it all together. My gut says its not needed.